patterns/1-initial/secure-innersource.md

```diff
@@ -0,0 +1,144 @@
+
+Restricting access typically reduces the risk of these things from happening, but at the same time hampers collaboration and re-use.
+
+# Context
+
+- The organization has an InnerSource Program Office (ISPO), or a similar group, steering the success of the InnerSource initiative in the organization. One of their goals is to stimulate maximum openness and transparency in the organization.
+- The organization has a Security Team constraining unnecessary data access to prevent the organization from data-leakage and malicious code injection.
+- The Security Team, ISPO, and engineering teams are not regularly talking with each other.
+- “Closed Source” is the default in the organization when creating new repositories i.e. only the team owning/maintaining the code has access to the given repo.
+- “Shared Source” within the organization isn’t common practice. Organizational teams aren’t familiar with what code or information should or shouldn’t be placed in shared repositories.
+
+# Forces
+
+- A successful InnerSource program needs openness and transparency (e.g. access to code, issues, documentation, and roadmap)
+- Good security practice is to minimize access, following the [principle of least privilege][principle-of-least-privilege].
+- The Security Team is more familiar with security principles at conceptual level than security practices at operation level. Some of the security policy is good at concept but hard to execute at operation level.
+- Engineering Teams focus more on service/product development or knowledge sharing than on measures for data protection. While it is easy for Engineering Teams to decide to close or open the repository, they are usually not willing to spend time on judging how to reach a balance between both.
+- No one-fit-all guideline or rules to judge what data or process is secured or not. That much depends on data sensitivity and overall security policy/infrastructure.
+
+# Solutions
+
+To reduce the misalignment and possible misunderstanding between the teams involved, it is key to bring everybody together so that they are able to express their goals and concerns, and to develop shared language together.
+
+After that they can decide as a group how to drive the execution of the specific goals, and what to do to reduce the risks that were identified.
+
+## Setup
+
+Start by bringing the Security Team, Engineering Teams and ISPO together, to discuss the goals of the InnerSource initiative, as well as any security concerns.
```

Comment:
From my experience, we also had to have one person from our legal team who we'd bring into these conversations as needed. The reason it had to be legal was, our internal data classification was owned by legal. So when our legal guidelines said "all software source code is confidential and therefore least privilege", we had to begin adding addendums and negotiate how to make exceptions to that classification.

Reply:
That makes sense. In some areas the concerns of the Security team and Legal team might even be overlapping.

Can you suggest a sentence that we could add? e.g. something like
> It may also be helpful to involve your Legal team in this phase, so that ...

Reply:
> It may also be helpful to involve your Legal team in this phase, so that any data classification documents can be surfaced and included in the scope of the InnerSource program's charter.

Reply:
Agree that including legal is a good thing, and it also helps in discussion with sceptics: "yes, we have people from Security and Legal on board".

Reply:
If I would read this without your prior statement I wouldn't get what "data classification documents" means. 
Is there a way to say this in different terms?

Reply:
```suggestion
Start by bringing (representatives if) Engineering, Security, Legal, and the ISPO together, to discuss the goals of the InnerSource initiative, as well as any security concerns.
```